### DISCUSSION TITLE
Support ingesting data from MAVEN2

### DISCUSSION THREAD

**Discussion by lparsons** [Princeton-LSI-ResearchComputing/tracebase#1057] [Ideas]
MAVEN2 is an update to the original MAVEN software and adds a number of new features, including natural abundance correction. The correction is convenient, but not terribly accurate, thus we anticipate many researchers will continue to use downstream correction algorithms prior to submission to TraceBase.

We will likely want to support the ability to import corrected data from MAVEN2, but that is a secondary concern to ensuring that MAVEN2 output can be processed through abundance correction and loaded into TraceBase.

@mneinast Can you upload a couple of MAVEN2 formatted files? We should test how well these can be used in AccuCor, IsoCorr, and IsoAutoCorr. If they work without modification to those programs, we can work on supporting those first.

1. Get copies of MAVEN2 output, both with and without corrected values
2. Test running MAVEN2 output through [AccuCor](https://cran.r-project.org/web/packages/accucor/index.html), [isocorr13C15N](https://github.com/xxing9703/isocorr13C15N), and [IsoAutocorr](https://github.com/xxing9703/Iso-Autocorr)
3. Post the results here.
    1. For any that are successful, define some issues to add support to TraceBase. This should include adding support for MAVEN2 corrected data.
    2. If there are issues running the data through any of the correction programs, we should define issues in the respective repositories. Depending on the nature of the issues, the fix may involve changes to the correction software, MAVEN2, or both.

**Comment by xxing9703** [comment]
[MAVEN2_S01_04_iso.csv](https://github.com/user-attachments/files/16289159/MAVEN2_S01_04_iso.csv)
I'm attaching here one example of MAVEN2 output with 13C and 15N checked.  Note the following format changes that will affect the downstream isotope correction:
1. column I, 'note'  used to be 'isotopeLabel'
2. column M, 'compoundFormula', used to be 'formula'
3. There are extra rows, where column 'note' is blank.  These rows serve as the summary for each compound, newly added in MAVEN2, need to be removed to avoid confusions.
4. In MAVEN2, all the columns with '(intensity)' in its name are sample intensity data block,  while MAVEN1 or ELMAVEN do not use such identified, but simply put all sample intensity data block right after the 'parent' column.  We can delete all columns without the (intensity) identifier after column 'parent'

After the above changes,  IsoAutocorr will work on MAVEN2 output.

After make
